code/modules/hallucinations/effects/moderate.dm

```diff
@@ -476,3 +476,23 @@
   */
 /obj/effect/hallucination/delusion/proc/get_image(mob/living/carbon/human/H)
 	return image('icons/mob/animal.dmi', H, pick("black_bear", "brown_bear", "corgi", "cow", "deer", "goat", "goose", "pig", "blank-body"))
+
+/obj/effect/hallucination/ventpeek
+	duration = 4 SECONDS
+
+/obj/effect/hallucination/ventpeek/Initialize(mapload, mob/living/carbon/hallucination_target)
+	. = ..()
+	
+	var/list/venttargets = list()
+	for(var/obj/machinery/atmospherics/unary/vent_pump/vent in oview(world.view, target))
+		venttargets += vent
+	if(!length(venttargets))
+		qdel(src)
+		return
+	var/image/I = image('icons/effects/effects.dmi', get_turf(pick(venttargets)))
+	add_icon(I)
+	flick("hallucination_clown", I)
```

Comment:
make sure this appears only to the hallucinating mob please

Reply:
Tested and updated testing section of PR.